### PR TITLE
create recipe for migmap

### PR DESCRIPTION
**Pull request by dkoppstein**
Currently, the tests for this PR are failing because of some issue with the test environment also relating to another package that I built, [igblast](https://github.com/bioconda/recipes/tree/master/recipes/igblast). MiGMAP depends on igblast. The igblast program in turn depends on an environmental variable IGDATA to point to where some required data live, so I defined it in etc/activate.d/ in the igblast recipe so that environmental variable is defined when the environment is sourced. However, this variable (and CONDA_ENV_PATH, the path of the conda environment, on which it depends) do not seem to be activated during the test phase of conda build. If I source activate the test or build directories and then run the programs manually, the programs work, but not during the test phase. 

If someone has time I think I'd appreciate a review of this PR and of my [igblast](https://github.com/bioconda/recipes/tree/master/recipes/igblast) recipe which also became more complicated than I first imagined. Sorry for asking so many questions, still getting the hang of this.

### CODE REVIEW

**Comment by dkoppstein** [comment]
C.f. conda/conda-build/#364

**Comment by johanneskoester** [comment]
Aaron Meurer says in the second post of that issue that relying on activate scripts is a bad idea. So I would say we should try something else. For example, you could wrap the commands that need environment variables into shell scripts that set them and forward all given arguments with ``$@``.

**Comment by dkoppstein** [comment]
I'm not sure if I agree with that philosophy. I feel that it is OK to execute the path/to/some/virtualenv/script because virtualenv is pure python and the virtualenv paths are hardcoded into the python binaries; however, for conda, I think you get into some dangerous territory -- for example, if the conda environment isn't sourced here, the java command will silently default to the system java, which is out of date and will cause errors. Of course, you could wrap the java command into the wrapper script too, but where does the complexity end? My instinct is that you should assume a conda environment is activated to run a program. 

For the purposes of getting it to work now, though, I've taken the approach to wrap the commands and will just grumble a bit =) I've updated the PR, which also includes an updated igblast. Again, I'd appreciate a quick review since I'm still getting started with this.

File: recipes/igblast/build.sh [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
 
 IGBLAST_ADDRESS=ftp://ftp.ncbi.nih.gov/blast/executables/igblast/release
-INCLUDE_DIR=$PREFIX/include/igblast
+SHARE_DIR=$PREFIX/share/igblast
 
 wget $IGBLAST_ADDRESS/edit_imgt_file.pl
-sed -i '1 s/^.*$/#!\/usr\/bin\/env perl/g' edit_imgt_file.pl # replace the first line with /usr/bin/env perl instead of the hard-coded /opt
+# replace the first line with /usr/bin/env perl instead of the hard-coded /opt
+# does this require an explicit perl dependency? 
+sed -i '1 s/^.*$/#!\/usr\/bin\/env perl/g' edit_imgt_file.pl
```
Comment (johanneskoester) [inline recipes/igblast/build.sh]: I always thought that conda is able to fix hardcoded paths in shebang files by itself. Doesn't this work here?
Apart from that, yes, the package should have a perl dependency then. But that is no problem, we already have perl in bioconda.

**Comment by johanneskoester** [comment]
Looks good apart from my little comment above.

**Comment by dkoppstein** [comment]
I just tried it without the sed command, it still seems to leave the hardcoded shebang. So I kept it and added the perl dependency as you suggested. I'll merge when the build is finished.

**Pull request merged**